components/sso-hostobject/org.wso2.carbon.hostobjects.sso/src/main/java/org/wso2/carbon/hostobjects/sso/SAMLSSORelyingPartyObject.java

```diff
@@ -402,14 +403,73 @@ public static String jsFunction_getUUID(Context cx, Scriptable thisObj,
-                                                       Function funObj)
+    public static String jsFunction_getSAMLAuthRequest(Context cx, Scriptable thisObj, Object[] args, Function funObj)
             throws Exception {
         SAMLSSORelyingPartyObject relyingPartyObject = (SAMLSSORelyingPartyObject) thisObj;
-        return Util.marshall(new AuthReqBuilder().
-                buildAuthenticationRequest(relyingPartyObject.getSSOProperty(SSOConstants.ISSUER_ID)));
-
+        boolean isPassiveRequired = false;
+        String acsUrl;
+        int argLength = args.length;
+        if (argLength == 2) {
+            isPassiveRequired = (Boolean) args[0];
+            if (isPassiveRequired) {
+                acsUrl = (String) args[1];
+                if (!Boolean.valueOf(relyingPartyObject.getSSOProperty(SSOConstants.SIGN_REQUESTS))) {
+                    //builds an unsigned passive authentication request
+                    return Util.marshall(new AuthReqBuilder().
+                            buildPassiveAuthenticationRequest(relyingPartyObject.getSSOProperty(SSOConstants.ISSUER_ID),
+                                    acsUrl));
+                } else {
+                    //builds a signed passive authentication request
+                    return Util.marshall(new AuthReqBuilder().
+                            buildPassiveSignedAuthenticationRequest(
+                                    relyingPartyObject.getSSOProperty(SSOConstants.ISSUER_ID),
+                                    MultitenantConstants.SUPER_TENANT_ID,
+                                    MultitenantConstants.SUPER_TENANT_DOMAIN_NAME,
+                                    relyingPartyObject.getSSOProperty(SSOConstants.IDP_URL),
+                                    acsUrl));
+                }
+            } else {
+                if (!Boolean.valueOf(relyingPartyObject.getSSOProperty(SSOConstants.SIGN_REQUESTS))) {
+                    //builds an unsigned non-passive authentication request
+                    return Util.marshall(new AuthReqBuilder().
+                            buildAuthenticationRequest(
+                                    relyingPartyObject.getSSOProperty(SSOConstants.ISSUER_ID)));
+                } else {
+                    //builds a signed non-passive authentication request
+                    return Util.marshall(new AuthReqBuilder().
+                            buildSignedAuthRequest(
+                                    relyingPartyObject.getSSOProperty(SSOConstants.ISSUER_ID),
+                                    MultitenantConstants.SUPER_TENANT_ID,
+                                    MultitenantConstants.SUPER_TENANT_DOMAIN_NAME));
+                }
+            }
+        } else {
+            //When number of args are not eq. to 2, then its assumed to be a non passive auth request build
+            if (!Boolean.valueOf(relyingPartyObject.getSSOProperty(SSOConstants.SIGN_REQUESTS))) {
```

Comment:
Fixed in #1534(https://github.com/wso2/carbon-apimgt/commit/80d5819bcc58795d672e6159c7f828df399458dc)